internal/cli/options.go

```diff
@@ -10,9 +10,12 @@ type Options struct {
 	Verbose bool
 	step.Factory
 	KubeconfigPath string
+	Finalizer      *Finalizer
```

Comment:
From the consumer perspective this contains 0 or more finalizers, not just 1. I would call the field `Finalizers`.
That way, when adding, it looks more natural: `opts.Finalizers.Add(...)`.

Reply:
good point!

Reply:
:)